src/libNode/FinalBlockProcessing.cpp

```diff
@@ -761,6 +762,11 @@ bool Node::ProcessFinalBlockCore(const bytes& message, unsigned int offset,
                                         .GetBlockNum() +
                                     1
                              << "] FINISH WRITE STATE TO DISK");
+        if (m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() <
```

Comment:
Putting `ENABLE_ACCOUNTS_POPULATING` check first will be even more efficient.
So we avoid stuff like `GetLastBlock()` having to acquire mutex, etc...

Reply:
Fixed for both places